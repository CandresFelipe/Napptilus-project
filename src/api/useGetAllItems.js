import { useQuery } from 'react-query';

import { appClient } from '../services/client';

export async function getAllItems() {
  const data = await appClient.get('/product');
  return data;
}

export function useGetAllItems() {
  return useQuery({
    queryKey: 'products',
    queryFn: async () => {
      const { data } = await getAllItems();
      return data;
    }
  });
}
