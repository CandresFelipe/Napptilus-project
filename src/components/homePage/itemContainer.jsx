import { Box, Card, CardContent, Grid, Typography } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { useNavigate } from 'react-router-dom';

import { Button } from '../../common/Elements/Button';
import { ClickableImage } from '../../common/Elements/ClickableImage';

const useStyles = makeStyles({
    card: {
        minWidth: 300,
        paddingBottom: '1.5em',
        paddingTop: '1.5em'
    },

    buttonBox: {
        display: 'flex',
        justifyContent: 'space-around',
        marginTop: 5
    }
});

export function ItemContainer({ imageURL, brand, model, price, id }) {
    const classes = useStyles();
    const navigate = useNavigate();

    const _onClickDetails = (event) => {
        event.preventDefault();
        navigate(`/details/${id}`);
    };

    return (
        <Grid item lg={4} xs={12}>
            <Card className={classes.card} role='columnheader'>
                <ClickableImage alt='phoneImage' src={imageURL} />
                <CardContent sx={{ padding: '0.1em' }}>
                    <Typography variant='h6' color='black'>
                        {`${brand} - ${model}`}
                    </Typography>
                    <Typography
                        variant='h6'
                        color='black'
                        sx={{ textTransform: 'lowercase' }}
                    >
                        {price !== '' ? `precio: ${price} â‚¬` : 'No disponible'}
                    </Typography>
                </CardContent>
                <Box component='div' className={classes.buttonBox}>
                    <Button
                        data-testid='button-details'
                        buttonType='primary'
                        variant='button'
                        title='Ver detalles'
                        onClick={(event) => _onClickDetails(event)}
                    />
                </Box>
            </Card>
        </Grid>
    );
}
