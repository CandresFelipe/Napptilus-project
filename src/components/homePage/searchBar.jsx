import SearchIcon from '@mui/icons-material/Search';
import { Box, InputBase } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { useCallback, useState } from 'react';

import { useDataContext } from '../../context/dataContext.jsx';

const useStyles = makeStyles({
  box: {
    display: 'flex',
    flexGrow: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginTop: 30,
    marginBottom: 30,
    position: 'relative',
    width: 'auto'
  },
  searchBarIcon: {
    marginLeft: 5,
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    color: 'rgb(0,0,0,0.6)'
  },
  input: {
    background: 'rgb(0,0,0,0.1)',
    opacity: 0.7,
    borderRadius: 5,
    padding: 5,
    paddingLeft: 35
  },
  coloredLine: {
    borderTop: '0.1em solid  rgb(0,0,0,0.1)'
  },
  container: {
    flexDirection: 'column',
    marginTop: 90,
    marginBottom: 5,
    justifyContent: 'center'
  }
});

export function SearchBar() {
  const classes = useStyles();
  const [filteredData, setFilteredData] = useDataContext();
  const [val, setVal] = useState('');

  const _onChange = useCallback(
    (event) => {
      const value = event.target.value;
      console.log(value);
      setVal(value);
      if (value !== '') {
        setFilteredData(
          filteredData.filter((item) => item.model.includes(value))
        );
      } else {
        setFilteredData([]);
      }
    },
    [filteredData, setFilteredData]
  );

  const ColoredLine = () => <div className={classes.coloredLine}></div>;
  return (
    <div className={classes.container}>
      <ColoredLine />
      <Box component='div' className={classes.box}>
        <div className={classes.searchBarIcon}>
          <SearchIcon />
        </div>
        <InputBase
          type='text'
          className={classes.input}
          placeholder='Search...'
          inputProps={{ 'aria-label': 'search' }}
          onChange={_onChange}
          value={val}
        />
      </Box>
      <ColoredLine />
    </div>
  );
}
