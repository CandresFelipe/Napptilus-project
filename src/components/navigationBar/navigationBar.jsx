import { AppBar, Grid, IconButton, Toolbar } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { useNavigate } from 'react-router-dom';

import { Breadcrumbs } from './breadcrumbs.jsx';
import { PurchaseCart } from './purchaseCart.jsx';

import napptilusLogo from '../../assets/napptilus_icon.png';

const useStyles = makeStyles({
    appBar: {
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)'
    },
    image: {
        borderRadius: 60,
        maxHeight: 100,
        minHeight: 50
    }
});

export function NavigationBar() {
    const classes = useStyles();
    const navigate = useNavigate();

    const _onNavigateHome = () => {
        navigate('/');
    };

    return (
        <AppBar component='nav' className={classes.appBar}>
            <Toolbar>
                <Grid container item lg={12} xs={12} alignItems='center'>
                    <Grid item lg={3} xs={4}>
                        <Breadcrumbs />
                    </Grid>
                    <Grid item lg={6} xs={4}>
                        <IconButton
                            sx={{ flexGrow: 1 }}
                            size='small'
                            edge='start'
                            onClick={_onNavigateHome}
                        >
                            <img
                                src={napptilusLogo}
                                className={classes.image}
                            />
                        </IconButton>
                    </Grid>
                    <Grid item lg={3} xs={4} textAlign='right'>
                        <PurchaseCart />
                    </Grid>
                </Grid>
            </Toolbar>
        </AppBar>
    );
}
