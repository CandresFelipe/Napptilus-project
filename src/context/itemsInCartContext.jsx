import { createContext, useContext, useEffect, useState } from 'react';
const ItemInCartContext = createContext(undefined);

export function useItemInCartContext() {
  const context = useContext(ItemInCartContext);
  if (context === undefined) {
    throw new Error('use useItemsInCartContext within a ItemInCartProvider');
  }
  return context;
}

function getInitialState() {
  const value = localStorage.getItem('cartItems');
  return value ? JSON.parse(value) : 0;
}

export function ItemInCartProvider({ children }) {
  const [items, setItems] = useState(getInitialState);

  useEffect(() => {
    localStorage.setItem('cartItems', JSON.stringify(items));
  }, [items]);

  return (
    <ItemInCartContext.Provider value={[items, setItems]}>
      {children}
    </ItemInCartContext.Provider>
  );
}
