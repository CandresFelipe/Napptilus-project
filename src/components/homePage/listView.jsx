import { Grid } from '@mui/material';

import { ItemContainer } from './itemContainer.jsx';

export function ListView({ filteredData }) {
    return (
        <Grid container item lg={12} spacing={6}>
            {filteredData &&
                filteredData.map((item, index) => (
                    <ItemContainer
                        role={index}
                        key={item.id}
                        imageURL={item.imgUrl}
                        brand={item.brand}
                        model={item.model}
                        price={item.price}
                        id={item.id}
                    />
                ))}
        </Grid>
    );
}
