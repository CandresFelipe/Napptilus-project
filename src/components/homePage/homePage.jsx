import { Grid, Typography } from '@mui/material';
import React, { useEffect } from 'react';

import { ListView } from './listView.jsx';
import { SearchBar } from './searchBar.jsx';

import { useGetAllItems } from '../../api/useGetAllItems';
import { Spinner } from '../../common/Elements/Spinner';
import { useDataContext } from '../../context/dataContext.jsx';
import { AxiosErrorHandler } from '../../utils/errors/axiosErrorHandler.jsx';

export function HomePage() {
    const {
        data: itemsData,
        isSuccess,
        isError,
        error,
        isLoading
    } = useGetAllItems();
    const [filteredData, setFilteredData] = useDataContext();

    useEffect(() => {
        if (itemsData) {
            setFilteredData(itemsData);
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [itemsData]);

    return isError ? (
        <AxiosErrorHandler error={error} />
    ) : isLoading ? (
        <Spinner />
    ) : isSuccess ? (
        <Grid container item lg={12} spacing={4} style={{ marginTop: 90 }}>
            <Grid item lg={12} xs={12}>
                <Typography
                    sx={{ typography: { lg: 'h1', xs: 'h3' } }}
                    color='black'
                    alt='title-home-page'
                >
                    CompraTuMovil.com
                </Typography>
                <SearchBar />
            </Grid>
            <ListView filteredData={filteredData} />
        </Grid>
    ) : (
        <React.Fragment></React.Fragment>
    );
}
