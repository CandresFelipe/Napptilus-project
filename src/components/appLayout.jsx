import { Route, Routes } from 'react-router-dom';

import { DetailsPage } from './detailPage';
import { HomePage } from './homePage';
import { NavigationBar } from './navigationBar';

import { DataProvider } from '../context/dataContext.jsx';
import { ItemInCartProvider } from '../context/itemsInCartContext.jsx';

export default function AppLayout() {
  return (
    <div>
      <DataProvider>
        <ItemInCartProvider>
          <NavigationBar />
          <Routes>
            <Route path='/' element={<HomePage />} />
            <Route path='/details/:id' element={<DetailsPage />} />
          </Routes>
        </ItemInCartProvider>
      </DataProvider>
    </div>
  );
}
