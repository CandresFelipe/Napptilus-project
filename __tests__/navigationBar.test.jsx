import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter } from 'react-router-dom';

import {
    mockDataItems,
    mockUseLocationValue,
    mockUseLocationValueWhenDetailsLanded
} from './__fixtures__/mockData';
import { dataWrapper } from './__fixtures__/wrappers.js';

import { useGetAllItems } from '../src/api/useGetAllItems.js';
import { HomePage } from '../src/components/homePage/homePage.jsx';
import { NavigationBar } from '../src/components/navigationBar';
import { ItemInCartProvider } from '../src/context/itemsInCartContext.jsx';

const queryClient = new QueryClient();

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
    useLocation: jest
        .fn()
        .mockImplementation(() => {
            return mockUseLocationValueWhenDetailsLanded;
        })
        .mockImplementationOnce(() => {
            return mockUseLocationValueWhenDetailsLanded;
        })
        .mockImplementationOnce(() => {
            return mockUseLocationValue;
        })
}));

const mockedUseGetAllItems = useGetAllItems;
jest.mock('../src/api/useGetAllItems.js');

const wrapper = ({ children }) => (
    <BrowserRouter>
        <QueryClientProvider client={queryClient}>
            <ItemInCartProvider>{children}</ItemInCartProvider>
        </QueryClientProvider>
    </BrowserRouter>
);

describe('Check navigationBar component renders correctly', () => {
    it('renders navigationBar', () => {
        const component = render(<NavigationBar />, { wrapper });
        expect(component).toMatchSnapshot();
    });
    describe('Check breadcrumbs lands correctly', () => {
        it('renders only the home link when is homePage', () => {
            render(<NavigationBar />, { wrapper });
            expect(screen.getByTestId('home-link-text')).toHaveTextContent(
                'Inicio'
            );
            expect(screen.queryByTestId('product-details-link')).toBeNull();
        });

        it('renders both, home and details breadcrumbs', () => {
            mockedUseGetAllItems.mockImplementation(() => ({
                data: mockDataItems,
                isSuccess: true
            }));
            render(<HomePage />, { wrapper: dataWrapper });
            screen.getAllByTestId('button-details')[0].onclick();
            render(<NavigationBar />, { wrapper });
            expect(screen.getByTestId('home-link')).toHaveTextContent('Inicio');
            expect(
                screen.getByTestId('product-details-link')
            ).toHaveTextContent('Detalles del Producto');
        });
    });
});
