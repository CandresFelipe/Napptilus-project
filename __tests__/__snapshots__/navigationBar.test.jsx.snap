// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check navigationBar component renders correctly renders navigationBar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed makeStyles-appBar-1 mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12 css-1x4vn2y-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-3 css-120v30d-MuiGrid-root"
            >
              <div
                class="makeStyles-breadcrumbs-3"
                role="presentation"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 MuiGrid-grid-lg-12 css-sa6qj6-MuiGrid-root"
                >
                  <nav
                    arial-aria-label="breadcrumbs"
                    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
                  >
                    <ol
                      class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                    >
                      <li
                        class="MuiBreadcrumbs-li"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-12 css-pivvrl-MuiGrid-root"
                        >
                          <a
                            class="makeStyles-link-4"
                            data-testid="home-link"
                            href="/"
                          >
                            Inicio
                          </a>
                        </div>
                      </li>
                      <li
                        aria-hidden="true"
                        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                      >
                        /
                      </li>
                      <li
                        class="MuiBreadcrumbs-li"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-12 css-pivvrl-MuiGrid-root"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-lzsq43-MuiTypography-root"
                            data-testid="product-details-link"
                          >
                            Detalles del Producto
                          </p>
                        </div>
                      </li>
                    </ol>
                  </nav>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-6 css-r0tab3-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeSmall css-1ynsyw7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <img
                  class="makeStyles-image-2"
                  src="[object Object]"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-3 css-eqjzfc-MuiGrid-root"
            >
              <button
                arial-label="purchase-cart"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ShoppingCartTwoToneIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m15.55 11 2.76-5H6.16l2.37 5z"
                      opacity=".3"
                    />
                    <path
                      d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-zkqb44-MuiBadge-badge"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed makeStyles-appBar-1 mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12 css-1x4vn2y-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-3 css-120v30d-MuiGrid-root"
          >
            <div
              class="makeStyles-breadcrumbs-3"
              role="presentation"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 MuiGrid-grid-lg-12 css-sa6qj6-MuiGrid-root"
              >
                <nav
                  arial-aria-label="breadcrumbs"
                  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
                >
                  <ol
                    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                  >
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-12 css-pivvrl-MuiGrid-root"
                      >
                        <a
                          class="makeStyles-link-4"
                          data-testid="home-link"
                          href="/"
                        >
                          Inicio
                        </a>
                      </div>
                    </li>
                    <li
                      aria-hidden="true"
                      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                    >
                      /
                    </li>
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-12 css-pivvrl-MuiGrid-root"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-lzsq43-MuiTypography-root"
                          data-testid="product-details-link"
                        >
                          Detalles del Producto
                        </p>
                      </div>
                    </li>
                  </ol>
                </nav>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-6 css-r0tab3-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeSmall css-1ynsyw7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <img
                class="makeStyles-image-2"
                src="[object Object]"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-3 css-eqjzfc-MuiGrid-root"
          >
            <button
              arial-label="purchase-cart"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ShoppingCartTwoToneIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m15.55 11 2.76-5H6.16l2.37 5z"
                    opacity=".3"
                  />
                  <path
                    d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-zkqb44-MuiBadge-badge"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
